File: algoExpert/Arrays/Easy/transposeMatrix.cpp
#include <vector>

using namespace std;

vector<vector<int>> transposeMatrix(vector<vector<int>> matrix) {
  // Write your code here.
  vector<vector<int>> flippedMatrix = {};
  for(int i = 0; i < matrix[0].size(); i++) {
    vector<int> currentRow = {};
    for(int j = 0; j < matrix.size(); j++) {
      currentRow.push_back(matrix[j][i]);
    }
    flippedMatrix.push_back(currentRow);
  }
  return flippedMatrix;
}

File: algoExpert/Arrays/Easy/transposeMatrix.py
def transposeMatrix(matrix):
    # Write your code here.
    tranposed = []
    for i in range(len(matrix[0])):
        getValuesFromMatrix(matrix, i, tranposed)
    return tranposed


def getValuesFromMatrix(matrix, idx, newArray):
    row = []
    for i in range(len(matrix)):
        row.append(matrix[i][idx])
    newArray.append(row)